app/main.py

```diff
@@ -1,2 +1,60 @@
 class Dictionary:
-    pass
+
+    def __init__(self):
+        self.capacity = 8
+        self.length = 0
+        self.t_hold = int(self.capacity * (2 / 3))
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def adding_item(self, key, value):
+        hash_index = hash(key) % self.capacity
+        if not self.hash_table[hash_index]:
+            self.hash_table[hash_index].extend([key, value])
+            self.length += 1
+        else:
+            i = 0
+            while i < len(self.hash_table):
+                if self.hash_table[i]:
+                    if self.hash_table[i][0] == key:
+                        self.hash_table[i] = [key, value]
+                        break
+                    else:
+                        i += 1
+                else:
+                    i += 1
```

Comment:
The same here, you don't have to use counter. Use index = (index + 1) % self.initial_capacity.